frame/support/src/traits/tokens/nonfungibles_v2.rs

```diff
@@ -0,0 +1,257 @@
+	///   successfully.
+	/// * `maybe_check_owner`: An optional account id that can be used to authorize the destroy
+	///   command. If not provided, we will not do any authorization checks before destroying the
+	///   item.
+	///
+	/// If successful, this function will return the actual witness data from the destroyed item.
+	/// This may be different than the witness data provided, and can be used to refund weight.
+	fn destroy(
+		collection: Self::CollectionId,
+		witness: Self::DestroyWitness,
+		maybe_check_owner: Option<AccountId>,
+	) -> Result<Self::DestroyWitness, DispatchError>;
+}
+
+/// Trait for providing an interface for multiple collections of NFT-like items which may be
+/// minted, burned and/or have attributes set on them.
+pub trait Mutate<AccountId, ItemConfig>: Inspect<AccountId> {
+	/// Mint some `item` of `collection` to be owned by `who`.
+	///
+	/// By default, this is not a supported operation.
+	fn mint_into(
+		_collection: &Self::CollectionId,
+		_item: &Self::ItemId,
+		_who: &AccountId,
+		_config: &ItemConfig,
+		_deposit_collection_owner: bool,
+	) -> DispatchResult {
+		Err(TokenError::Unsupported.into())
+	}
+
+	/// Burn some `item` of `collection`.
+	///
+	/// By default, this is not a supported operation.
+	fn burn(
+		_collection: &Self::CollectionId,
+		_item: &Self::ItemId,
+		_maybe_check_owner: Option<&AccountId>,
+	) -> DispatchResult {
+		Err(TokenError::Unsupported.into())
+	}
+
+	/// Set attribute `value` of `item` of `collection`'s `key`.
+	///
+	/// By default, this is not a supported operation.
+	fn set_attribute(
+		_collection: &Self::CollectionId,
+		_item: &Self::ItemId,
+		_key: &[u8],
+		_value: &[u8],
+	) -> DispatchResult {
+		Err(TokenError::Unsupported.into())
+	}
+
+	/// Attempt to set the strongly-typed attribute `value` of `item` of `collection`'s `key`.
+	///
+	/// By default this just attempts to use `set_attribute`.
+	fn set_typed_attribute<K: Encode, V: Encode>(
+		collection: &Self::CollectionId,
+		item: &Self::ItemId,
+		key: &K,
+		value: &V,
+	) -> DispatchResult {
+		key.using_encoded(|k| value.using_encoded(|v| Self::set_attribute(collection, item, k, v)))
+	}
+
+	/// Set attribute `value` of `collection`'s `key`.
+	///
+	/// By default, this is not a supported operation.
+	fn set_collection_attribute(
+		_collection: &Self::CollectionId,
+		_key: &[u8],
+		_value: &[u8],
+	) -> DispatchResult {
+		Err(TokenError::Unsupported.into())
+	}
+
+	/// Attempt to set the strongly-typed attribute `value` of `collection`'s `key`.
+	///
+	/// By default this just attempts to use `set_attribute`.
+	fn set_typed_collection_attribute<K: Encode, V: Encode>(
+		collection: &Self::CollectionId,
+		key: &K,
+		value: &V,
+	) -> DispatchResult {
+		key.using_encoded(|k| {
+			value.using_encoded(|v| Self::set_collection_attribute(collection, k, v))
+		})
+	}
+}
+
+/// Trait for providing a non-fungible sets of items which can only be transferred.
```

Comment:
Same question. Should it be?:
```suggestion
/// Trait for transferring non-fungible sets of items.
```

Reply:
Shall we remove the `set of items` at all?

Reply:
Well I think on the trait doc it can say `set`, because there could be another function that does actually transfer a set. Just for the one function there it transfers a single item.